### PR TITLE
Adding panic-handler and allocator features to `icu_capi`

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]
r=me but I'd also wait for @sffc's approval

**Review by Manishearth — DISMISSED** [review]

File: docs/tutorials/cpp.md [hunk 1]
```diff
@@ -105,24 +109,11 @@ C++ versions beyond C++17 are supported, as are other C++ compilers.
...
-
-```rust
-#![no_std]
-
-// Expose icu_capi symbols
-extern crate icu_capi;
-
-#[global_allocator]
-static ALLOCATOR: dlmalloc::GlobalDlmalloc = dlmalloc::GlobalDlmalloc;
+Users wishing to use ICU4X on a `no_std` platform will need to provide an allocator and a panic hook in order to build a linkable library. The `icu_capi` crate can provide a looping panic handler, and a `malloc`-backed allocator, under the `panic-handler` and `libc-alloc` features, respectively.
```
Comment (sffc) [inline docs/tutorials/cpp.md:112]: Question: how does panic-abort and panic-immediate-abort work with this new built-in panic handler?

File: ffi/capi/Cargo.toml [hunk 2]
```diff
@@ -104,6 +104,12 @@ compiled_data = [
     "icu_timezone?/compiled_data",
 ]
 
+std = []
+# Adds a looping panic handler (useful for no_std targets)
+panic-handler = []
+# Uses malloc as the global allocator (useful for no_std targets)
+libc-alloc = ["dep:libc_alloc"]
```
Comment (sffc) [inline ffi/capi/Cargo.toml:111]: Nits on feature names:

1. It seems we're using `_` for feature names so we should be consistent
2. `panic-handler` should be more specific, like `looping_panic_handler`

**Review by sffc — COMMENTED** [review]
Seems good

**Review by sffc — APPROVED** [review]

**Pull request merged**